src/garvis/heartbeat_service.py

```
@@ -188,40 +303,55 @@ def learn(state: OABState) -> None:
+                if state.status is CycleStatus.FAILED:
+                    self.rebound_count += 1
+                    wait_seconds = backoff
+                    backoff = min(30.0, backoff * 2.0)
+                else:
+                    backoff = max(
+                        0.1,
+                        self.interval_seconds or 0.1,
+                    )
             except KeyboardInterrupt:
                 raise
             except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)